File: examples/python/sdk_demo.py
#!/usr/bin/env python3
"""SDK demo: quote + unsigned prepare. No keys."""
from fastflip import FastFlip

ff = FastFlip()
listed = ff.markets(status="open", limit=5)
market = next((m for m in listed["markets"] if m.get("open")), None)
if not market:
    print("no open markets")
    raise SystemExit(0)
mid = market["id"]
print("market", mid, market.get("question"))
print("quote", ff.quote(mid, 0.01, side="yes"))
print("unsigned tx", ff.prepare_buy(mid, 0.01, side="yes")["tx"])

File: examples/javascript/sdk-demo.mjs
import { FastFlip } from "../../sdk/js/dist/index.js";

const ff = new FastFlip();
const list = await ff.markets({ status: "open", limit: 5 });
const id = list.markets.find((m) => m.open)?.id ?? list.markets[0]?.id;
console.log("open count", list.count, "using market", id);
if (id == null) process.exit(0);
const q = await ff.quote({ marketId: id, side: "yes", eth: 0.01 });
console.log("quote cents", q.avgCents, "shares", q.shares);
const prep = await ff.prepareBuy({ marketId: id, side: "yes", eth: 0.01 });
console.log("unsigned", prep.tx);
